scripts/generate-commitid.ps1

```diff
@@ -3,5 +3,16 @@
 
 param($OutDir)
 
+$file_name = ($OutDir + "\" + "git_commit_id.h")
 $commit_id = git rev-parse HEAD
-'#define GIT_COMMIT_ID "' + $commit_id + '"' + "`n" | out-file -FilePath ($OutDir + "\\" + "git_commit_id.h")
\ No newline at end of file
+$new_content = '#define GIT_COMMIT_ID "' + $commit_id + '"' + "`n"
+[string]$old_commit_id = ""
+if (Test-Path $file_name) {
+    [string]$old_content = ""
+    $old_content = Get-Content $file_name
+    $old_commit_id = $old_content.Split('"')[1]
+}
+if ($old_commit_id -ne $commit_id) {
```

Comment:
Why not just 
```
if ($old_content -ne $new_content) {
```
and then you don't need to extract the old_commit_id

Reply:
Because it doesn't work (for reasons I was unable to determine in a reasonable amount of time). Out-File and Get-Content results in some subtle changes to the string resulting in them not matching.

Reply:
Ok, this is not blocking.